django/forms/widgets.py

```diff
@@ -680,6 +680,27 @@ def get_context(self, name, value, attrs=None):
             context['widget']['attrs']['multiple'] = 'multiple'
         return context
 
+    def use_required_attribute(self, initial):
+        """
+        Don't render 'required' if the first <option> has a value, as that's
+        invalid HTML.
+        """
+        use_required_attribute = super(Select, self).use_required_attribute(initial)
+        # 'required' is always okay for <select multiple>.
+        if self.allow_multiple_selected:
+            return use_required_attribute
+
+        def choice_has_empty_value(choice):
+            """Return True if the choice's value is empty string or None."""
+            value, _ = choice
+            return (
+                (isinstance(value, six.string_types) and not bool(value)) or
+                value is None
+            )
```

Comment:
Might be worth defining a non local function for this purpose as it's going to be used on every form rendering?